pkg/kv/kvserver/replica_consistency.go

```diff
@@ -665,7 +673,14 @@ func (r *Replica) computeChecksumPostApply(
 		}
 		// NB: the names here will match on all nodes, which is nice for debugging.
 		tag := fmt.Sprintf("r%d_at_%d", r.RangeID, as.RaftAppliedIndex)
-		if dir, err := r.store.checkpoint(ctx, tag); err != nil {
+		// If the node is about to be killed, create a full checkpoint, otherwise
+		// create a smaller partial checkpoint including this Replica and maybe a
+		// few neighbouring replicas.
+		desc := &desc
+		if shouldFatal {
+			desc = nil
```

Comment:
PS: I think we should take the same checkpoint across all nodes. On the terminated node we still have the entire database available if we need to look at anything else.

Reply:
We wouldn't be able to easily take the **same** checkpoint (i.e. with matching begin/end key), since different nodes may have different replica sets. But we can do the same algorithm: checkpoint the range and its neighbours.

Reply:
>We wouldn't be able to easily take the same checkpoint (i.e. with matching begin/end key), since different nodes may have different replica sets. But we can do the same algorithm: checkpoint the range and its neighbours.

Sure, that's fine -- this is mostly useful for Pebble-level stuff, where the LSM structure is going to be different across nodes anyway.

Reply:
Done. Using the same checkpoint algorithm regardless of the node termination.